c2h/include/c2h/check_results.cuh

```diff
@@ -26,23 +26,153 @@
  ******************************************************************************/
 #pragma once
 
+#include <cub/detail/type_traits.cuh>
+
+#include <cuda/std/complex>
 #include <cuda/std/type_traits>
 
-#include "c2h/catch2_test_helper.h"
+#include <catch2/matchers/catch_matchers_floating_point.hpp>
+#include <test_util.h>
 
-/**
- * @brief Compares the results returned from system under test against the expected results.
- */
-// TODO: Replace this function by REQUIRE_APPROX_EQ once it supports integral vector types like short2
 template <typename T>
-void verify_results(const c2h::host_vector<T>& expected_data, const c2h::device_vector<T>& test_results)
+void verify_results(const c2h::host_vector<T>& expected_data, const c2h::host_vector<T>& test_results)
 {
-  if constexpr (::cuda::std::is_floating_point_v<T>)
+  using namespace cub::detail;
+  int device_id                = 0;
+  int compute_capability_major = 0;
+  int compute_capability_minor = 0;
+  CubDebugExit(cudaGetDevice(&device_id));
+  CubDebugExit(cudaDeviceGetAttribute(&compute_capability_major, cudaDevAttrComputeCapabilityMajor, device_id));
+  CubDebugExit(cudaDeviceGetAttribute(&compute_capability_minor, cudaDevAttrComputeCapabilityMinor, device_id));
+  int compute_capability = 10 * compute_capability_major + compute_capability_minor;
```

Comment:
I think `cub::PtxVersion` is more appropriate here, because we need to check what compute capability the kernels were compiled for, not what the GPU supports. Or do I miss something here?